src/System.Security.Cryptography.Encryption.Aes/src/Internal/Cryptography/AesCngCryptoDecryptor.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+
+using Internal.NativeCrypto;
+
+namespace Internal.Cryptography
+{
+    internal sealed class AesCngCryptoDecryptor : AesCngCryptoTransform
+    {
+        public AesCngCryptoDecryptor(CipherMode cipherMode, PaddingMode paddingMode, byte[] key, byte[] iv, int blockSize)
+            : base(cipherMode, paddingMode, key, iv, blockSize)
+        {
+        }
+
+        protected sealed override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                byte[] heldoverCipher = _heldoverCipher;
+                _heldoverCipher = null;
+                if (heldoverCipher != null)
+                {
+                    Array.Clear(heldoverCipher, 0, heldoverCipher.Length);
+                }
+            }
+
+            base.Dispose(disposing);
+        }
+
+        protected sealed override void Reset()
+        {
+            base.Reset();
+            if (_heldoverCipher != null)
+            {
+                Array.Clear(_heldoverCipher, 0, _heldoverCipher.Length);
```

Comment:
i don't know the answer to this, but there's an ArrayT<T>.Clear in the ArrayT.cs file that a bunch of projects bring in...  We should follow-up to see what the rhyme or reason is to when to use that vs Array.Clear as is done here.